bittide/tests/Tests/DoubleBufferedRam.hs

```diff
@@ -885,3 +891,169 @@ wbStorageProtocolsModel = property $ do
+
+    footnote $ "simTransactionsA" <> show simTransactionsA
+    footnote $ "simTransactionsB" <> show simTransactionsB
+    footnote $ "wbRequestsA" <> show wbRequestsA
+    footnote $ "wbRequestsB" <> show wbRequestsB
+    footnote $ "goldenTransactionsA" <> show goldenTransactionsA
+    footnote $ "goldenTransactionsB" <> show goldenTransactionsB
+
+    assert $ and (L.zipWith compareTransWithUndef simTransactionsA goldenTransactionsA)
+    assert $ and (L.zipWith compareTransWithUndef simTransactionsB goldenTransactionsB)
+
+   where
+    genWishboneTransfer ::
+      (KnownNat addressWidth, KnownNat (BitSize a)) =>
+      Int -> -- ^ size
+      Gen a ->
+      Gen (WishboneMasterRequest addressWidth a, Int)
+    genWishboneTransfer size genA =
+      let
+        validAddr = (4 *) . fromIntegral <$> Gen.enum 0 (size - 1)
+        -- Make wbOps that won't be repeated
+        mkRead address bs = (Read address bs, 0)
+        mkWrite address bs a = (Write address bs a, 0)
+      in
+        -- Generate valid operations. The boolean represents the validity of the operation.
+      Gen.choice
+        [ (mkRead  <$> validAddr <*> genDefinedBitVector)
+        , (mkWrite <$> validAddr <*> genDefinedBitVector <*> genA)
+        ]
+
+    checkAddressEqual :: (KnownNat addressWidth, KnownNat (BitSize a)) =>
+        WishboneMasterRequest addressWidth a ->
+        WishboneMasterRequest addressWidth a ->
+        Bool
+    checkAddressEqual req1 req2 = extractAddress req1 == extractAddress req2
+        where
+            extractAddress :: WishboneMasterRequest addressWidth a -> BitVector addressWidth
+            extractAddress (Read addr _) = addr
+            extractAddress (Write addr _ _) = addr
+
+    getAddressWbRequest :: (KnownNat addressWidth, KnownNat (BitSize a)) =>
+        WishboneMasterRequest addressWidth a -> Unsigned addressWidth
+    getAddressWbRequest (Read addr _) = unpack addr
+    getAddressWbRequest (Write addr _ _) = unpack addr
+
+    getByteEnableWbRequest :: (KnownNat addressWidth, KnownNat (BitSize a)) =>
+        WishboneMasterRequest addressWidth a -> Unsigned (BitSize a `DivRU` 8)
+    getByteEnableWbRequest (Read _ be) = unpack be
+    getByteEnableWbRequest (Write _ be _) = unpack be
+
+-- | Memory Model
+type MemoryModel a b = Map.Map a b
+
+-- | Behavioral model for 'wbStorageTDPM'.
+wbStorageTDPBehaviorModel ::
+  forall addrW bytes .
+  ( 1 <= addrW, KnownNat bytes) =>
+  (KnownNat addrW) =>
+  [WishboneMasterRequest addrW (Bytes bytes)] -> [WishboneMasterRequest addrW (Bytes bytes)] ->
+  [(Transaction addrW bytes (Bytes bytes),Transaction addrW bytes (Bytes bytes))]
+wbStorageTDPBehaviorModel initWbOpsA initWbOpsB = case (cancelMulDiv @bytes @8) of
+  Dict -> snd $ L.mapAccumL g emptyMemoryMap (L.zipWith (\wbA wbB -> (wbA, wbB)) initWbOpsA initWbOpsB)
+   where
+    emptyMemoryMap = Map.empty :: MemoryModel (BitVector addrW) (Bytes bytes)
```

Comment:
I think just using `Map` is fine here.

It does not seem to contribute to the clarity of the API, nor does it make your code more concise.

Reply:
Agree